File: translations/codenet/mistral_commenetd/C2J/deepseek/Code/s479404625.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Declare three String variables to store strings A, B, and C
        String A = "", B = "", C = "";

        // Initialize the String variables A, B, and C using scanner.next() function
        A = scanner.next();
        B = scanner.next();
        C = scanner.next();

        // Calculate the length of strings A and B using length() method and store the lengths in la and lb respectively
        int la = A.length(), lb = B.length();

        // Logical block to check if the last character of string A is same as the first character of string B
        // and the last character of string B is same as the first character of string C
        if (A.charAt(la - 1) == B.charAt(0) && B.charAt(lb - 1) == C.charAt(0)) {
            // If the condition is true, print YES on the console
            System.out.println("YES");
        }
        // Logical block to check if the condition in if block is false
        else {
            // If the condition is false, print NO on the console
            System.out.println("NO");
        }

        // Return 0 to indicate successful execution of the program
    }
}

File: translations/codenet/mistral_commenetd/C2C++/granite/Code/s479404625.cpp


#include <iostream>
#include <string>

// Declare three string variables A, B, and C
std::string A, B, C;

int main()
{
	// Initialize the string variables A, B, and C using cin function
	std::cin >> A >> B >> C;

	// Calculate the length of strings A and B using size function and store the lengths in la and lb respectively
	int la = A.size(), lb = B.size();

	// Logical block to check if the last character of string A is same as the first character of string B
	// and the last character of string B is same as the first character of string C
	if (A[la - 1] == B[0] && B[lb - 1] == C[0])
	{
		// If the condition is true, print YES on the console
		std::cout << "YES\n";
	}
	// Logical block to check if the condition in if block is false
	else
	{
		// If the condition is false, print NO on the console
		std::cout << "NO\n";
	}

	// Return 0 to indicate successful execution of the program
	return 0;
}

